include/dmlc/logging.h

```diff
@@ -35,6 +35,17 @@ struct Error : public std::runtime_error {
 };
 
 #if DMLC_LOG_STACK_TRACE
+// get stack trace logging depth from env variable.
+inline size_t LogStackTraceLevel() {
+  size_t level;
+  if (auto var = std::getenv("DMLC_LOG_STACK_TRACE_DEPTH")) {
+    if (1 == sscanf(var, "%zu", &level)) {
+      return level + 1;
+    }
+  }
+  return 0;
```

Comment:
```suggestion
  return DMLC_LOG_STACK_TRACE_SIZE;
```

@szha I think this change will preserve existing behavior